### DISCUSSION TITLE
Large inrush

### DISCUSSION THREAD

**Discussion by pretenda** [jmcollin78/solar_optimizer#73] [Ideas]
First up, thanks for an awesome integration. I've been using it now for a few weeks and it's been so much fun watching my devices turn on/off based on how much sun there is. Such a fantastic experience. 

For my question, I've got a heat pump that draws a large amount of electricity to start with (6kW), but then kicks down to 2kW or so once it's been running for about 10 minutes. What I'd love is for it to only start if that 6kW is available, but to continue running even if there's only 2kW free once it's up and running. Is there a way to take this into account? I'm finding if I have the max current set to 6kW, it turns off more often than it needs to, and if it's set to 2kW it draws from the grid on startup when I'd rather it just wait. 

Happy if I'm just doing something wrong, just would love to be steered in the right direction. Thanks heaps!

**Comment by jmcollin78** [comment]
Hello, this is not possible out of the box. You can always try to tweak the `check_usable_template` to implement complex rules but it could be hard to do and you will need extra helper entities to store the state.
